acurl/acurl/__init__.py

```diff
@@ -379,7 +381,11 @@ def json(self):
     @property
     def headers(self):
         if not hasattr(self, '_headers'):
-            self._headers = dict(self.headers_tuple)
+            self._headers = CaseInsensitiveDefaultDict(list)
+            for k, v in self.headers_tuple:
+                self._headers[k].append(v)
+            for k in self._headers:
+                self._headers[k] = ', '.join(self._headers[k])
```

Comment:
This is maybe slightly funky in that we're writing lists to self._headers and then rewriting them as strings, which might fly against the expectations set on line 384. Kind of relates to my first comment on the PR and could be worked around by applying the same change.

Reply:
I did read your comment and though it was good in the first iteratios but I don't know why I just felt it was unpalatable to have multiple instances of this new type of dict floating about, purely because it needs that behaviour in both loops(maybe?). Although only the first one needs to be based off default dict I think? But that involves me writing more code to have two types of case insenstive ones.

However I will make two if people think that's better.

Reply:
Ooh, yeah, I think having two is going to be better (with the first a local).  Only the second actually needs to be case-insensitive (and need not inherit from defaultdict either...)

Reply:
That's fair. Having to have the case sensitivity logic in two places isn't something I'd considered.

Reply:
No I think you're wrong about that @aecay, I think both need to be case insensitive as the header_tuple could contain lower and upper case 'keys'. I know the right thing is to have one that inherits from default and one from dict but I'm just lazy, I'll go do it.

Reply:
🤦 you're right.